tmk_core/protocol/chibios/usb_main.c

```diff
@@ -692,96 +310,60 @@ static bool usb_request_hook_cb(USBDriver *usbp) {
-                                osalSysUnlockFromISR();
-                            }
                         }
                         usbSetupTransfer(usbp, NULL, 0, NULL);
                         return TRUE;
                         break;
 
                     case HID_SET_IDLE:
-                        keyboard_idle = usbp->setup[3]; /* MSB(wValue) */
-                                                        /* arm the timer */
-#ifdef NKRO_ENABLE
-                        if (!keymap_config.nkro && keyboard_idle) {
-#else  /* NKRO_ENABLE */
-                        if (keyboard_idle) {
-#endif /* NKRO_ENABLE */
-                            osalSysLockFromISR();
-                            chVTSetI(&keyboard_idle_timer, 4 * TIME_MS2I(keyboard_idle), keyboard_idle_timer_cb, (void *)usbp);
-                            osalSysUnlockFromISR();
-                        }
-                        usbSetupTransfer(usbp, NULL, 0, NULL);
-                        return TRUE;
-                        break;
+                        return usb_set_idle_cb(usbp);
                 }
                 break;
         }
     }
 
     /* Handle the Get_Descriptor Request for HID class (not handled by the default hook) */
-    if ((usbp->setup[0] == 0x81) && (usbp->setup[1] == USB_REQ_GET_DESCRIPTOR)) {
+    if (((usbp->setup[0] & (USB_RTYPE_DIR_MASK | USB_RTYPE_RECIPIENT_MASK)) == (USB_RTYPE_DIR_DEV2HOST | USB_RTYPE_RECIPIENT_INTERFACE)) && (usbp->setup[1] == USB_REQ_GET_DESCRIPTOR)) {
         dp = usbp->config->get_descriptor_cb(usbp, usbp->setup[3], usbp->setup[2], get_hword(&usbp->setup[4]));
         if (dp == NULL) return FALSE;
         usbSetupTransfer(usbp, (uint8_t *)dp->ud_string, dp->ud_size, NULL);
         return TRUE;
     }
 
-    for (int i = 0; i < NUM_USB_DRIVERS; i++) {
-        if (drivers.array[i].config.int_in) {
-            // NOTE: Assumes that we only have one serial driver
-            return qmkusbRequestsHook(usbp);
+    for (int i = 0; i < USB_ENDPOINT_IN_COUNT; i++) {
+        if (usb_endpoints_in[i].usb_requests_cb != NULL) {
+            if (usb_endpoints_in[i].usb_requests_cb(usbp)) {
+                return true;
+            }
         }
     }
 
-    return FALSE;
+    return false;
 }
 
-/* Start-of-frame callback */
-static void usb_sof_cb(USBDriver *usbp) {
-    osalSysLockFromISR();
-    for (int i = 0; i < NUM_USB_DRIVERS; i++) {
-        qmkusbSOFHookI(&drivers.array[i].driver);
-    }
-    osalSysUnlockFromISR();
+static __attribute__((unused)) void dummy_cb(USBDriver *usbp) {
+    (void)usbp;
 }
 
-/* USB driver configuration */
 static const USBConfig usbcfg = {
     usb_event_cb,          /* USB events callback */
     usb_get_descriptor_cb, /* Device GET_DESCRIPTOR request callback */
-    usb_request_hook_cb,   /* Requests hook callback */
-    usb_sof_cb             /* Start Of Frame callback */
+    usb_requests_hook_cb,  /* Requests hook callback */
+#if STM32_USB_USE_OTG1 == TRUE
```

Comment:
I don't know how common it is, but would it be worth checking for otg2, as well? 
Mostly, just noticed that a lot of devices had it, when trying to figure out where the issue was popping up for me.

Reply:
Thanks, added it to the check. Not sure if it is affected as well, as it is the High Speed capable iteration of the device, but won't hurt either.

Reply:
From the brief skimming, it should support FS, but requires the extra stuff for HS.  (but I could be wrong).